config/commands.js

```diff
@@ -676,18 +676,22 @@ var commands = exports.commands = {
 					break;
 
 				case 'moves':
+					var moves = {};
+					for (var i in searches[search]) {
+						moves[i] = Tools.getMove(i);
+						if (!moves[i].exists) return this.sendReplyBox("'" + moves[i] + "' is not a known move.");
+					}
 					for (var mon in dex) {
 						var template = Tools.getTemplate(dex[mon].id);
 						if (!template.learnset) template = Tools.getTemplate(template.baseSpecies);
 						if (!template.learnset) continue;
-						for (var i in searches[search]) {
-							var move = Tools.getMove(i);
-							if (!move.exists) return this.sendReplyBox("'" + move + "' is not a known move.");
+						for (var i in moves) {
+							var moveId = moves[i].id;
 							var prevoTemp = Tools.getTemplate(template.id);
```

Comment:
I'll compile a list of searches to try with the cached prevoTemps to see the actual impact that has, because it slows the search down slightly in cases of only searching one move.